src/thread_task.rs

```diff
@@ -804,13 +794,11 @@ pub unsafe fn rav1d_worker_task(c: &Rav1dContext, task_thread: Arc<Rav1dTaskCont
                 }
             }
             // run decoding tasks last
-            while ttd.cur.load(Ordering::Relaxed) < c.n_fc {
-                let first_0 = ttd.first.load(Ordering::SeqCst);
-                let fc = &mut *(c.fc).offset(
-                    first_0
-                        .wrapping_add(ttd.cur.load(Ordering::Relaxed))
-                        .wrapping_rem(c.n_fc) as isize,
-                );
+            while (ttd.cur.load(Ordering::Relaxed) as usize) < c.fc.len() {
+                let fc = c
+                    .fc_iter(ttd.first.load(Ordering::SeqCst) as usize)
+                    .nth(ttd.cur.load(Ordering::Relaxed) as usize)
+                    .unwrap();
```

Comment:
I think `%` is probably more efficient here than `.nth` on `Chain`, and we use `%` and indexing in the other cases.  `.fc_iter` is better when we iterate through all of them.

```suggestion
                let first = ttd.first.load(Ordering::SeqCst);
                let fc = &c.fc[(first + ttd.cur.load(Ordering::Relaxed)) as usize % fc.len()];
```

Reply:
Efficiency with very small numbers of elements here isn't going to matter, but sure.